From 78a098f220c8354d14f42875455b0b2b016c2465 Mon Sep 17 00:00:00 2001
From: softwarenoob <admin@softwarenoob.com>
Date: Sun, 14 Nov 2021 03:35:37 +0200
Subject: [PATCH 1/3] fix(openj9): migrate to IBM official openj9 images

AdoptOpenJDK are deprecated and removed. They are no longer maintained for legal licensing reasons after moving to Eclipse. IBM Semeru continues to maintain the images with the JDK including openj9.
---
 java/11j9/Dockerfile | 2 +-
 java/16j9/Dockerfile | 2 +-
 java/8j9/Dockerfile  | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/java/11j9/Dockerfile b/java/11j9/Dockerfile
index ca6d9a0..8325fee 100644
--- a/java/11j9/Dockerfile
+++ b/java/11j9/Dockerfile
@@ -20,7 +20,7 @@
 # SOFTWARE.
 #
 
-FROM        --platform=$TARGETOS/$TARGETARCH adoptopenjdk/openjdk11-openj9:debianslim
+FROM        --platform=$TARGETOS/$TARGETARCH ibm-semeru-runtimes:open-11-jdk
 
 LABEL       author="Matthew Penner" maintainer="matthew@pterodactyl.io"
 
diff --git a/java/16j9/Dockerfile b/java/16j9/Dockerfile
index e1b5579..125b73d 100644
--- a/java/16j9/Dockerfile
+++ b/java/16j9/Dockerfile
@@ -20,7 +20,7 @@
 # SOFTWARE.
 #
 
-FROM        --platform=$TARGETOS/$TARGETARCH adoptopenjdk/openjdk16-openj9:debianslim
+FROM        --platform=$TARGETOS/$TARGETARCH ibm-semeru-runtimes:open-16-jdk
 
 LABEL       author="Matthew Penner" maintainer="matthew@pterodactyl.io"
 
diff --git a/java/8j9/Dockerfile b/java/8j9/Dockerfile
index 3cfdd10..616ad69 100644
--- a/java/8j9/Dockerfile
+++ b/java/8j9/Dockerfile
@@ -20,7 +20,7 @@
 # SOFTWARE.
 #
 
-FROM        --platform=$TARGETOS/$TARGETARCH adoptopenjdk/openjdk8-openj9:debianslim
+FROM        --platform=$TARGETOS/$TARGETARCH ibm-semeru-runtimes:open-8-jdk
 
 LABEL       author="Matthew Penner" maintainer="matthew@pterodactyl.io"
 

From 61db1995ccb4f8128016d2a3f6c0af3879256f96 Mon Sep 17 00:00:00 2001
From: softwarenoob <admin@softwarenoob.com>
Date: Sun, 14 Nov 2021 03:36:14 +0200
Subject: [PATCH 2/3] ci(openj9): enable workflows

---
 .github/workflows/java.yml | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/.github/workflows/java.yml b/.github/workflows/java.yml
index 5b7e5af..1a55f66 100644
--- a/.github/workflows/java.yml
+++ b/.github/workflows/java.yml
@@ -17,11 +17,11 @@ jobs:
       matrix:
         tag:
           - 8
-          #- 8j9
+          - 8j9
           - 11
-          #- 11j9
+          - 11j9
           - 16
-          #- 16j9
+          - 16j9
           - 17
     steps:
       - uses: actions/checkout@v2

From 3a157a3e7dd57210324c3801e5ccca712daafa76 Mon Sep 17 00:00:00 2001
From: softwarenoob <admin@softwarenoob.com>
Date: Tue, 14 Dec 2021 20:39:40 +0200
Subject: [PATCH 3/3] feat: java openj9 image

---
 .github/workflows/java.yml |  2 +-
 README.md                  |  4 +++-
 java/17j9/Dockerfile       | 39 ++++++++++++++++++++++++++++++++++++++
 3 files changed, 43 insertions(+), 2 deletions(-)
 create mode 100644 java/17j9/Dockerfile

diff --git a/.github/workflows/java.yml b/.github/workflows/java.yml
index 1a55f66..a989b2e 100644
--- a/.github/workflows/java.yml
+++ b/.github/workflows/java.yml
@@ -21,8 +21,8 @@ jobs:
           - 11
           - 11j9
           - 16
-          - 16j9
           - 17
+          - 17j9
     steps:
       - uses: actions/checkout@v2
       - uses: docker/setup-qemu-action@v1
diff --git a/README.md b/README.md
index cbb7358..0faa80a 100644
--- a/README.md
+++ b/README.md
@@ -19,7 +19,7 @@ example of this would be something like Java or Python which are used for runnin
 All of these images are available for `linux/amd64` and `linux/arm64` versions, unless otherwise specified, to use
 these images on an arm64 system, no modification to them or the tag is needed, they should just work.
 
-### Contributing
+## Contributing
 
 When adding a new version to an existing image, such as `java v42`, you'd add it within a child folder of `java`, so
 `java/42/Dockerfile` for example. Please also update the correct `.github/workflows` file to ensure that this new version
@@ -61,6 +61,8 @@ is tagged correctly.
     * `ghcr.io/pterodactyl/yolks:java_16j9`
   * [`java17`](https://github.com/pterodactyl/yolks/tree/master/java/17)
     * `ghcr.io/pterodactyl/yolks:java_17`
+  * [`java17 - OpenJ9`](https://github.com/pterodactyl/yolks/tree/master/java/17j9)
+    * `ghcr.io/pterodactyl/yolks:java_17j9`
 * [`nodejs`](https://github.com/pterodactyl/yolks/tree/master/nodejs)
   * [`node12`](https://github.com/pterodactyl/yolks/tree/master/nodejs/12)
     * `ghcr.io/pterodactyl/yolks:nodejs_12`
diff --git a/java/17j9/Dockerfile b/java/17j9/Dockerfile
new file mode 100644
index 0000000..9539224
--- /dev/null
+++ b/java/17j9/Dockerfile
@@ -0,0 +1,39 @@
+#
+# Copyright (c) 2021 Matthew Penner
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+FROM --platform=$TARGETOS/$TARGETARCH ibm-semeru-runtimes:open-17-jdk
+
+LABEL author="Matthew Penner" maintainer="matthew@pterodactyl.io"
+
+LABEL org.opencontainers.image.source="https://github.com/pterodactyl/yolks"
+LABEL org.opencontainers.image.licenses=MIT
+
+RUN apt-get update -y &&
+	apt-get install -y curl ca-certificates openssl git tar sqlite3 fontconfig libfreetype6 tzdata iproute2 libstdc++6 &&
+	useradd -d /home/container -m container
+
+USER container
+ENV USER=container HOME=/home/container
+WORKDIR /home/container
+
+COPY ./../entrypoint.sh /entrypoint.sh
+CMD [ "/bin/bash", "/entrypoint.sh" ]
